BUILDING.md

```diff
@@ -29,40 +29,44 @@ By default, this tarball does not contain a Phoenix client jar as it is meant to
 of Phoenix version (one PQS release can be used against any Phoenix version). Today, PQS builds against
 the Phoenix 4.15.0-HBase-1.4 release.
 
+Note that the resulting Query Server binaries are not tied to any Phoenix, Hbase or Hadoop versions,
+and will work with all recent Phoenix versions.
+
 ```
-$ mvn package
+$ mvn clean package
 ```
 
 ### Bundling a Phoenix Client
 
 To build a release of PQS which packages a specific version of Phoenix, specify the `package-phoenix-client` system property
-and specify the `phoenix.version` system property to indicate a specific Phoenix version.
-
-PQS will package the same version of Phoenix used for build/test. This version is controlled by the
- `phoenix.version` system property.
+and specify the `phoenix.version` system property to indicate a specific Phoenix version, as well as
+the `phoenix.client.artifactid` to choose the phoenix-client HBase variant for 4.16+ / 5.1+.
 
 ```
-$ mvn package -Dpackage.phoenix.client -Dphoenix.version=5.1.0-SNAPSHOT
+$ mvn clean package -Dpackage.phoenix.client -Dphoenix.version=5.1.0-SNAPSHOT -Dphoenix.client.artifactid=phoenix-client-hbase-2.4
 ```
 
 ### Running integration tests
 
 `mvn package` will run the unit tests while building, but it will not run the integration test suite.
 
-The IT suite is run when executing `mvn install` or `mvn verify`. The Phoenix version specified
-with `phoenix.version` is used for running the integration tests.
+The IT suite is run when executing `mvn install` or `mvn verify`. The Phoenix client artifact specified
+with `phoenix.version` and `phoenix.client.artifactid` is used for running the integration tests.
 
-When specifying `phoenix.version`, also specify the HBase version to be used
-for integration testing by activating the corresponding `hbase<minor.major>` profile.
+When specifying `phoenix.version` and `phoenix.client.artifactid`, also specify the HBase version to
+be used for integration testing by activating the corresponding `hbase<minor.major>` profile.
 
-When using a Phoenix 5.x version, activate the `hbase-2.x` profile in addition to the
+When using a Phoenix 5.1+ version, activate the `hbase-2.x` profile in addition to the
 profile for the minor 2.x version.
 
+**NOTE** that the integration tests cannot be currently run with Phoenix 4.16+ or 5.1+, so the
+examples below will fail. See https://issues.apache.org/jira/browse/PHOENIX-63241
```

Comment:
typo
I think it should be https://issues.apache.org/jira/browse/PHOENIX-6324

Reply:
Thank you @richardantal

Reply:
Fixed